apis/hive/v1/azure/platform.go

```diff
@@ -24,6 +24,10 @@ type Platform struct {
 	// If empty, the value is equal to "AzurePublicCloud".
 	// +optional
 	CloudName CloudEnvironment `json:"cloudName,omitempty"`
+
+	// ResourceGroupName specifies a resource group where the cluster will be installed.
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
We deliberately decided not to do this when designing #1997 because we didn't want to have to validate/reconcile potential differences between a new hive API field and what the user put in the install-config. We decided the install-config is the source of truth, and in the normal provisioning path we accept the value from the installer right next to the InfraID in clusterMetadata.